server.go

```diff
@@ -378,21 +378,27 @@ func (s *Server) handleQuestion(q dns.Question, resp *dns.Msg, query *dns.Msg, i
 		return nil
 	}
 
-	switch q.Name {
-	case s.service.ServiceTypeName():
-		s.serviceTypeName(resp, s.ttl)
-		if isKnownAnswer(resp, query) {
-			resp.Answer = nil
+	if q.Qtype == dns.TypeA || q.Qtype == dns.TypeAAAA {
```

Comment:
I think it would be cleaner to add "case s.service.HostName:" and put any conditionals inside, so you don't need to reindent the rest of the code.

Reply:
Thought the same.